scipy/special/_complexstuff.pxd

```diff
@@ -82,6 +89,14 @@ cdef inline number_t zsin(number_t x) nogil:
     else:
         return libc.math.sin(x)
 
+cdef inline number_t zcos(number_t x) nogil:
+    cdef np.npy_cdouble r
+    if number_t is double_complex:
+        r = npy_ccos((<np.npy_cdouble*>&x)[0])
+        return (<double_complex*>&r)[0]
```

Comment:
This is actually wrong, as noticed in gh-6074
Needs to be changed to use npy_cdouble_from_double_complex and double_complex_from_npy_cdouble